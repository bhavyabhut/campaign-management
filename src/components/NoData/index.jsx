import React from 'react';
import style from './noData.module.css';

const NoData = () => (
  <>
    <div className={style.NoData}>
      <span>No Campaign Data Found</span>
    </div>
  </>
);

export default NoData;
