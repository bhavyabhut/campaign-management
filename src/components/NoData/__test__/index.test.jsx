import React from 'react';
import ReactDom from 'react-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NoData from '../index';
import renderer from 'react-test-renderer';

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<NoData />, div);
});

it('No Data Text Testing', () => {
  const { container } = render(<NoData />);
  expect(container.querySelector('div span')).toHaveTextContent('No Campaign Data Found');
});

it('No Data Found match snapshot', () => {
  const tree = renderer.create(<NoData />).toJSON();
  expect(tree).toMatchSnapshot();
});
