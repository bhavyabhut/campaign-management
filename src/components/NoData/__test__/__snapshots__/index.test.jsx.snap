// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Data Found match snapshot 1`] = `
<div
  className="NoData"
>
  <span>
    No Campaign Data Found
  </span>
</div>
`;
